src/components/_skeleton.scss

```diff
@@ -0,0 +1,74 @@
+@mixin Manifold__Skeleton {
+  @keyframes Manifold__Skeleton__Animation {
+    to {
+      transform: translateX(100%); // line width shouldn’t be more than this
+    }
+  }
+
+  border-radius: 2px;
+  color: transparent;
+  display: inline-block;
+  line-height: 1;
+  overflow: hidden;
+  position: relative;
+
+  &::before {
+    background-image: linear-gradient(
+      rgba(0, 0, 0, 0.1),
+      rgba(0, 0, 0, 0.1) 1em,
+      rgba(0, 0, 0, 0) 1em,
+      rgba(0, 0, 0, 0) 1.4em
+    );
+    background-repeat: repeat-y;
+    background-size: 100% 1.4em;
+    bottom: 0;
+    content: '';
+    left: 0;
+    pointer-events: none;
+    position: absolute;
+    right: 0;
+    top: 0;
+  }
+
+  &::after {
+    animation-duration: 1.5s;
+    animation-iteration-count: infinite;
+    animation-name: Manifold__Skeleton__Animation;
+    animation-timing-function: linear;
+    background-image: linear-gradient(
+      60deg,
+      rgba(255, 255, 255, 0) 50%,
+      rgba(255, 255, 255, 0.5),
+      rgba(255, 255, 255, 0) 60%
+    );
+    background-repeat: no-repeat;
+    content: '';
+    height: 100%;
+    opacity: 0.4;
+    pointer-events: none;
+    position: absolute;
+    right: 100%;
+    top: 0;
+    width: 60em; // derived from font size
+  }
+}
+
+@mixin Manifold__Skeleton--White {
+  &::before {
+    background-image: linear-gradient(
+      rgba(255, 255, 255, 0.25),
+      rgba(255, 255, 255, 0.25) 1em,
+      rgba(255, 255, 255, 0) 1em,
+      rgba(255, 255, 255, 0) 1.4em
+    );
+  }
+
+  &::after {
+    background-image: linear-gradient(
+      60deg,
+      rgba(0, 0, 0, 0) 50%,
+      rgba(0, 0, 0, 0.375),
+      rgba(0, 0, 0, 0) 60%
+    );
+  }
+}
```

Comment:
@ntassone in case you’re interested in possibly tweaking this, this would be a good opportunity!

Currently all the Sass in `components/` is manually-maintained. Partially because there are things we may need that may not be in a Figma component; partially because certain things like animations are kinda impossible to generate from Figma at the moment.

Anyway if you wanted to give input on this style in particular, that’d be awesome.

Reply:
Just check out https://mercury-fypuwqucn.now.sh/?path=/story/components--skeleton

Questions would include:

1. Should we add another style for light gray backgrounds?
2. How do you feel about the corner radii dropping off for multi-line (it’s hard to simulate with linear gradients)?
3. How does the animation feel in general (it feels a little robotic to me, but can’t find a good example of something I like better other than “strobing”—dimming opacity subtly)?

Reply:
![2020-03-31 15-58-39 2020-03-31 15_59_00](https://user-images.githubusercontent.com/1369770/78079039-8dbd2e00-7368-11ea-9608-73f93576346d.gif)

Alternate animation: flickering

Reply:
👍 Looking great to me.

I like the combination animation that's in storybook now – the dimming paired with the gradient swipe. 

Re: radii, do you think there's a way to generate a nested element (`span`?) and put the linear gradient on each span? I agree that the radii on the multi line feels a bit weird, it'd be ideal if it was on each line.

Reply:
> do you think there's a way to generate a nested element (span?) and put the linear gradient on each span?

Yeah that’s a good suggestion! But there’s actually not a simple way to do that. Because we’d have to know the # of lines, and that’s document flow. We could only get that by virtually rendering it offcanvas, or something else convoluted. But also, using JS in general is outside what Mercury can do; I’m trying to limit this to what a single CSS class can accomplish.

We could just kill it altogether and it would at least be consistent 🤔 it’s even hard to make a “only have corner radii on one-liners” because again that’s document flow and gets into responsive design, etc.